scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/PreferCurlyFors.scala

```diff
@@ -24,70 +33,49 @@ object PreferCurlyFors extends Rewrite {
   *     b <- bs if b > 2
   *   } yield (a, b)
   */
-class PreferCurlyFors(implicit ctx: RewriteCtx) extends RewriteSession {
+private class PreferCurlyFors extends FormatTokensRewrite.Rule {
 
-  import ctx.dialect
-  import ctx.tokenTraverser._
+  import FormatTokensRewrite._
+  import PreferCurlyFors._
 
-  private def findForParens(forTokens: Tokens): Option[(Token, Token)] =
-    for {
-      forToken <- forTokens.find(_.is[Token.KwFor])
-      leftParen <- findAfter(forToken) {
-        case _: Token.LeftParen => Some(true)
-        case Whitespace() => None
-        case _ => Some(false)
-      }
-      rightParen <- ctx.getMatchingOpt(leftParen)
-    } yield (leftParen, rightParen)
+  override def enabled(implicit style: ScalafmtConfig): Boolean =
+    PreferCurlyFors.enabled
```

Comment:
Isn't this just always `true` ?

Reply:
for this specific rule, yes. but depending on `ScalafmtConfig` (which can potentially be overridden locally), it could actually be generally enabled (otherwise wouldn't be created) but not apply in some cases.